src/Scripting/CSharpTest/ScriptTests.cs

```diff
@@ -178,6 +178,75 @@ public void Do()
 , ScriptOptions.Default.WithReferences(MscorlibRef, SystemRef, SystemCoreRef, CSharpRef));
         }
 
+        [WorkItem(6676, "https://github.com/dotnet/roslyn/issues/6676")]
+        [Fact]
+        public void TestRunEmbeddedStatementNotFollowedBySemicolon()
```

Comment:
>TestRunEmbeddedStatementNotFollowedBySemicolon [](start = 20, length = 46)

Please confirm that we have a test for a similar scenario in a non-script context.
Also, please add a test suggested by @gafter:
```
    if (e) a = b // duplicate "missing ';'" error given here
    throw e;
```